From 7807f4338957838f66a19f758d723089192518d9 Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Mon, 8 Aug 2016 15:55:20 -0400
Subject: [PATCH 0722/1722] drm/amd/dal: do not reset underlay pipe in
 apply_ctx_to_hw

Underlay reset is handled in commit surface, so it should be left
alone in apply_ctx_to_hw. This fixes a crash when changing mode
during S3

Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index 4907d58..9775902 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1363,7 +1363,8 @@ static enum dc_status apply_ctx_to_hw(
 		 * PHY when not already disabled.
 		 */
 
-		if (!pipe_ctx_old->stream)
+		/* Skip underlay pipe since it will be handled in commit surface*/
+		if (!pipe_ctx_old->stream || pipe_ctx_old->top_pipe)
 			continue;
 
 		if (!pipe_ctx->stream ||
-- 
2.7.4

